woocommerce/woocommerce
# Use exact match when comparing normalized path against normalized pag…

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #52895  

This PR fixes a bug where a URL containing part of a store page's URL is incorrectly identified as a valid store page URL.

I might be oversimplifying it, but I couldn’t find a reason to use `strpos`. I’ve replaced it with `===`, and all unit tests are passing.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch.
2. Set the site to Coming soon mode with `Apply to store pages only` option.
3. Create a new page from `Pages -> Add a new page`
4. Set the page title to `Shopping URL` and confirm the permalink is set to `shopping-url`.
5. Open the page in an incognito session.
6. The page content should render.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

 Fixes a bug where a URL containing part of a store page's URL is incorrectly identified as a valid store page URL

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
